### DISCUSSION TITLE
An idea for Plasma Profile (Potentially others)

### DISCUSSION THREAD

**Discussion by DarkXero-dev** [archlinux/archinstall#2484] [Ideas]
Hey there...

In the spirit of making ArchInstall relevant I cam up with this idea. For now it applies to Plasma, but with some effort could be implemented for other profiles as well. 

Let me be frank here, biggest complaint I and others have is with how current profiles work... Some are missing essential elements with some not being very consistent.. That's why I had to come up with a better more forward thinking solution.. 

![image](https://github.com/archlinux/archinstall/assets/81101439/0d5d7a09-f1c7-4d0b-8556-32dd40575e94)

If not my way, then I could help with optimizing existing ones keep them as they are while being consistent, and making them more functional. I am a Plasma user that's the DE I know best but with help from some friends I can gather more info for Gnome and Hyprland. But you get the idea.

I sure hope this doesn't fall on deaf ears, and we can collaborate on making ArchInstall more functional...

**Comment by svartkanin** [comment]
The point of archinstall is to install a minimum working system for any flavor of DE/WM. Having multiple options for each profile will add a  lot of complexity and a lot of maintenance of all these packages.

**Reply by DarkXero-dev** [reply]
Maybe. That's why I offered to help make current ones better keeping in line with ArchInstall's ideology. As is profiles are inconsistent and incomplete. And am not the only one saying this.

https://www.youtube.com/watch?v=XTGGx6ted14

**Comment by svartkanin** [comment]
Could you give a quick overview of how updating the profiles, specifically the packages would look like? 
I don't think having multiple options per profile is going to be complex, but potentially having a more all-rounded profiles would be a way forward. The major goal should be to distinguish subjective preferences over objective usability though. 
@Torxed probably has some input here as well :)

**Comment by DarkXero-dev** [comment]
I agree.. My method might be a bit overkill.. As for the all rounded profiles, since I am more experienced with plasma, I have done many tests and in the picture I shared, the "XeroLinux" profile is what resulted in a more complete and balanced experience without it being bloated.. 

As for the Wayland/WMs as Brodie said in video I shared, would be great if consistency was observed, like using same terminal and bars when possible for example.. And for Hyprland although current profile "works", not well, would be a good idea to include a few more packages that offer a more functional system OTB. 

Finally I would like to mention issue with nVidia drivers in ArchInstall, I did so before and I still insist that the way it's currently being done is incomplete.. The way it should be for DKMS and Open-DKMS is like this (Lines 49 -> 64)  [**Here**](https://github.com/DarkXero-dev/Storage/blob/2814f9b2c269bb073b14a680018c1f56c4f56f7e/xero-plasma.sh#L49)

Otherwise not everything will work.. 

Thanks for listening btw ;)

**Comment by svartkanin** [comment]
I agree that consistency would be great across profiles. There's a PR open to add some core packages to Hyprland https://github.com/archlinux/archinstall/pull/2481, as I'm not using Hyprland I can't fully judge if anything else is missing or could be added as "essential". 

You're welcome to raise a PR with a more curated list of packages - maybe as a start for Plasma - to start. I can't guarantee that it will go in but it good start a good discussion point if the arguments for certain packages are valid.
